### ISSUE TITLE
Button to upload to Strava

### ISSUE THREAD

**Issue by marcoahp** [scubajorgen/TomTomWatch#32]
Great Job scubajorgen!!!  The app works perfectly. Thanks for your effort, you have a 9.9. You'd have 10 if you include a button to upload the trk file to Strava. I upload the files manually and it works like a charm.

**Comment by scubadunc** [comment]
I agree with the above feature request and additionally:
1. Buttons for automatically uploading activities to Strava, MapMyFitness, Runkeeper incl. segments elevations etc
2. Buttons to export to fit and tcx formats...gpx (already implemented)

This would make it a 10/10

**Comment by scubajorgen** [comment]
Hum... It is fairly complex to upload to Strava because of the OAuth authentication process from a desktop application. This requires interaction with a browser.
This is out of scope for this app I am afraid.
I'll look at other file types. I guess TCX is easier than FIT.
Auto download GPX could be done...

**Comment by marcoahp** [comment]
Upload to Strava could be done with https://gotoes.org/gotoes/strava/
With this web you can upload files to Strava and Garmin easily, join gpx files and even upload to Strava and Garmin by email.

**Comment by scubadunc** [comment]
> Hum... It is fairly complex to upload to Strava because of the OAuth authentication process from a desktop application. This requires interaction with a browser. This is out of scope for this app I am afraid. I'll look at other file types. I guess TCX is easier than FIT. Auto download GPX could be done...

That would be amazing.  The problem I guess is that the gpx file exported doesn't contain heart rate info and calories, whereas a tcx does?

**Comment by marcoahp** [comment]
I upload gps to Strava and yes, they contain heart rate.

 

De: scubadunc ***@***.***> 
Enviado el: lunes, 9 de octubre de 2023 02:53
Para: scubajorgen/TomTomWatch ***@***.***>
CC: marcoahp ***@***.***>; Author ***@***.***>
Asunto: Re: [scubajorgen/TomTomWatch] Button to upload to Strava (Issue #32)

 

Hum... It is fairly complex to upload to Strava because of the OAuth authentication process from a desktop application. This requires interaction with a browser. This is out of scope for this app I am afraid. I'll look at other file types. I guess TCX is easier than FIT. Auto download GPX could be done...

That would be amazing. The problem I guess is that the gpx file exported doesn't contain heart rate info and calories, whereas a tcx does?

**Comment by scubajorgen** [comment]
GPX as exported can contain heartrate in the extensions. See [issue #36](https://github.com/scubajorgen/TomTomWatch/issues/36)

**Comment by mike5x3** [comment]
Upload to Strava could be done with https://github.com/RobbieLD/StravaUploader

**Comment by scubajorgen** [comment]
> Upload to Strava could be done with https://github.com/RobbieLD/StravaUploader

Does this mean that you can use this to sync files from a directory? This means, if you write a GPX using TomTomWatch to this directory that it automatically gets uploaded to Strava? If this is true, I'll write this solution to the FAQ :-)

**Comment by marcoahp** [comment]
Yes, and you could launch stravauploader.exe from your app, to download from watch and upload to strava with the same button

 

De: Jorgen van der Velde ***@***.***> 
Enviado el: viernes, 3 de noviembre de 2023 15:34
Para: scubajorgen/TomTomWatch ***@***.***>
CC: marcoahp ***@***.***>; Author ***@***.***>
Asunto: Re: [scubajorgen/TomTomWatch] Button to upload to Strava (Issue #32)

 

Upload to Strava could be done with https://github.com/RobbieLD/StravaUploader

Does this mean that you can use this to sync files from a directory? This means, if you write a GPX using TomTomWatch to this directory that it automatically gets uploaded to Strava? If this is true, I'll write this solution to the FAQ :-)

**Comment by mike5x3** [comment]
> > Upload to Strava could be done with https://github.com/RobbieLD/StravaUploader
> 
> Does this mean that you can use this to sync files from a directory? This means, if you write a GPX using TomTomWatch to this directory that it automatically gets uploaded to Strava? If this is true, I'll write this solution to the FAQ :-)

Yes.
Configure the working gpx directory of TomTomWatch in the appsettings.json of StravaUploader:
  "Device": {
    "Name": "Volume",
    "Path": "\\TomTomWatch\\working\\gpx",
    "FileType": "GPX"
As well you need to configure in the appsettings.json the client id and the client secret from the API page of Strava (you'll need to sign up to the Strava API for this).
As soon as StravaUploader is started the gpx files will be automatically uploaded to Strava and deleted in the working gpx directory.

Just another thing:
I would be nice to have the type of workout in the exported gpx file.
Otherwise this needs to be done in Strava manually after the upload.
As a workaround I currently use some kind of post-processing with a shell script for the exported gpx files:
C:\TomTomWatch\working\gpx>
ls *.gpx | xargs -l perl -pi -w -e "s/\/name\>/\/name\>\<type\>cycling\<\/type\>/"
